### DISCUSSION TITLE
Multiple page elements on a single PDF page

### DISCUSSION THREAD

**Discussion by severmat52** [QuestPDF/QuestPDF#289] [Q&A]
I would like to arrange multiple Page elements within a single page, is there any way to do this?

For example, I would like to arrange 4 3.5x5.5 pages on a single 11x14 PDF page.

**Comment by bennetbo** [comment]
This is definitely not possible currently. Im curious, whats your use case for this?

**Reply by severmat52** [reply]
Ok, thanks for the reply. The use case is for printing on paper that has 4 separate sections. Each page element would be a different section, but the PDF page would contain 4 sections. I would like to use the Page element for this because of the need for a header and footer.

Edit. I realize there might be other PDF libraries that can do this, but I was more curious if there was functionality I was not seeing so this could be achieved directly in QuestPDF.

**Reply by bennetbo** [reply]
Well the `Page` element just uses the `Decoration` element internally (to generate header and footer). Using the decoration API you can create something like this:
```csharp
Document
    .Create(container =>
    {
        container.Page(page =>
        {
            page.Size(PageSizes.A4.Landscape());
            page.Margin(2, Unit.Centimetre);
            page.DefaultTextStyle(x => x.FontSize(20));

            page.Content()
                .PaddingVertical(1, Unit.Centimetre)
                .AlignCenter()
                .AlignMiddle()
                .Grid(x =>
                {
                    x.Columns(2);
                    x.Spacing(20);

                    //Size of each section.
                    float size = (float)(PageSizes.A4.Landscape().Height / 3);

                    void Section(IContainer c, string color, string text)
                    {
                        c.Height(size)
                         .Background(color)
                         .Decoration(d =>
                         {
                             d.Before().AlignCenter().Text("Header").FontColor(Colors.White);
                         
                             d.Content()
                              .ExtendVertical()
                              .ExtendHorizontal()
                              .AlignCenter()
                              .AlignMiddle()
                              .Text(text).FontColor(Colors.White);
                         
                             d.After().AlignCenter().Text("Footer").FontColor(Colors.White);
                         });
                    }

                    Section(x.Item(), Colors.Red.Darken1, "1");
                    Section(x.Item(), Colors.Blue.Darken1, "2");
                    Section(x.Item(), Colors.Green.Darken1, "3");
                    Section(x.Item(), Colors.Yellow.Darken1, "4");
                });
        });
    })
```

Result:
![image](https://user-images.githubusercontent.com/53836821/180274242-36a8a9d9-3409-433b-8f72-0bd9503f3f2d.png)

Hope it helps 😄

**Reply by severmat52** [reply]
Thank you for the detailed example and the quick reply, this is very helpful!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
